tests/cpp/test_pointwise.cpp

```diff
@@ -478,12 +478,106 @@ TEST_F(PointwiseTest, VIssue1567ectorizationFactorAnalysisCase3) {
   fe.compileFusion(fusion, aten_inputs, lparams);
   auto cg_outputs = fe.runFusion(aten_inputs, lparams);
 
-  EXPECT_EQ(params->vectorize, true);
+  EXPECT_TRUE(params->vectorize);
   EXPECT_EQ(params->unroll_factor, 2);
   EXPECT_TRUE(hasVectorizationCache(tv0));
   EXPECT_TRUE(hasVectorizationCache(tv1));
 
   testValidate(fusion, cg_outputs, aten_inputs, __LINE__, __FILE__);
 }
 
+// params: use 1D scheduler, sharded dimenson
+class ShardedPointwiseTest
+    : public NVFuserTest,
+      public testing::WithParamInterface<std::tuple<bool, int>> {};
+
+TEST_P(ShardedPointwiseTest, DID_Compatible) {
+  auto [use_1D_scheduler, sharded_dim] = GetParam();
+  auto fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  TensorView* tv0 = makeContigTensor(4);
+  TensorView* tv1 = makeContigTensor(2);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = add(tv0, tv0);
+  auto tv3 = broadcast(tv1, {true, true, false, false});
+  auto tv4 = add(tv2, tv3);
+  fusion->addOutput(tv4);
+
+  DeviceMesh mesh = DeviceMesh::createForNumDevices(4);
+  for (TensorView* tv : {tv0, tv2, tv3, tv4}) {
+    tv->setDeviceMesh(mesh);
+    tv->axis(sharded_dim)->parallelize(ParallelType::DIDx);
+  }
+  tv1->setDeviceMesh(mesh);
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
+  fec.profile(true);
+
+  // Trigger the 1D scheduler by using small input size.
+  std::vector<int64_t> input_size(3);
+  if (use_1D_scheduler) {
+    input_size = {16, 8, 24};
+  } else {
+    input_size = {1024, 32, 64};
+  }
```

Comment:
> I can probably break this up but there are two main things I want to test

Thanks for clarifying your intention and I like how you proposed to separate the concerns. So how about splitting this test into two? 
1. Create a sharded fusion and an unsharded fusion, getPointwiseHeuristics both with various input sizes, assert the returned PointwiseParams are the same. 
2. Create a sharded fusion, manually construct PointwiseParams with various `break_point`s, compileFusion, runFusion, and testValidate. This only needs one input size IIUC, because we will be manually set `break_point`.